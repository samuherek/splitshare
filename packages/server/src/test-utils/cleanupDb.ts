import { getConnection } from 'typeorm';

export default async function cleanupDb() {
  await getConnection().query('DELETE FROM bill_user');
  await getConnection().query('DELETE FROM receipt');
  await getConnection().query('DELETE FROM bill_invite');
  await getConnection().query('DELETE FROM bill');
  await getConnection().query('DELETE FROM "user"');
}
