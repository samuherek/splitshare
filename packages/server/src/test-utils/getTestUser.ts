import { Connection } from 'typeorm';
import { User } from '../features/user/entity';

export const testUser = {
  email: 'user@email.com',
};

export async function getTestUser(conn: Connection) {
  const user = await conn.getRepository(User).findOne(testUser);

  if (!user) {
    throw new Error('No Test user found');
  }

  return user;
}
