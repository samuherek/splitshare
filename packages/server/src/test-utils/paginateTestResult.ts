import { PaginationConnection } from '../types';

export default function paginateTestResult(
  resArray: any[]
): PaginationConnection<any> {
  return {
    edges: resArray.map((r) => ({ node: r, cursor: 'nodeCursor' })),
    pageInfo: {
      hasNextPage: false,
      endCursor: 'endCursor',
      itemsCount: 0,
    },
  };
}
