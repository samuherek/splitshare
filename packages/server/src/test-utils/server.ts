import { ApolloServer } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import { getSchema } from '../schema';

export default async function createTestServer(ctx: any) {
  const schema = await getSchema();

  const server = new ApolloServer({
    schema,
    mockEntireSchema: false,
    mocks: true,
    context: () => ctx,
  });

  return createTestClient(server);
}
