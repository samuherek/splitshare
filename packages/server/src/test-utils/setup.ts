// @ts-ignore
import * as tryToCatch from 'try-to-catch';
import { Connection } from 'typeorm';
import { connectTestDatabase } from './testConnection';

(async () => {
  const [error, conn]: [Error, Connection] = await tryToCatch(
    connectTestDatabase,
    { drop: true }
  );

  if (error) {
    throw new Error(error.message);
  }
})();
