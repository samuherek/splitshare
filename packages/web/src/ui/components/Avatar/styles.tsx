import styled from 'styled-components';

export const AvatarBase = styled.div<{ size: number }>`
  width: ${props => `${props.size}px`};
  height: ${props => `${props.size}px`};
  display: flex;
  overflow: hidden;
  position: relative;
  align-items: center;
  flex-shrink: 0;
  user-select: none;
  justify-content: center;

  .AvatarImg {
    width: 100%;
    height: 100%;
    object-fit: cover;
    text-align: center;
  }
`;

export const Circle = styled(AvatarBase)`
  border-radius: 50%;
`;
