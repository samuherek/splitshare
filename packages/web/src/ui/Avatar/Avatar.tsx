import clsx from "clsx";
import React from "react";
import { AvatarBase, Circle } from "./styles";

interface Props {
  alt: string;
  children?: any;
  className?: string;
  component?: any;
  imgProps?: object;
  sizes?: string;
  size?: number;
  src?: string;
  srcSet?: string;
  variant?: "circle" | "square";
}

export const classes = {
  root: "Avatar",
  img: "AvatarImg",
  circle: "circle",
  square: "square"
};

const themedComponent = {
  square: AvatarBase,
  circle: Circle
};

const Avatar = React.forwardRef<Props, any>(function Avatar(props, ref) {
  const {
    alt,
    children: childrenProp,
    className,
    component: componentProp,
    imgProps,
    sizes,
    src,
    srcSet,
    variant = "circle",
    size = 40,
    ...rest
  } = props;

  let children = null;
  const img = src || srcSet;

  if (img) {
    children = (
      <img
        alt={alt}
        src={src}
        srcSet={srcSet}
        sizes={sizes}
        className={classes.img}
        {...imgProps}
      />
    );
  } else {
    children = childrenProp;
  }

  // @ts-ignore
  let Component = themedComponent[variant] || AvatarBase;

  if (componentProp) {
    Component = componentProp;
  }

  return (
    <Component
      ref={ref}
      size={size}
      className={clsx(className, classes.root, {
        [classes.circle]: variant === "circle",
        [classes.square]: variant === "square"
      })}
      {...rest}
    >
      {children}
    </Component>
  );
});

export default Avatar;
