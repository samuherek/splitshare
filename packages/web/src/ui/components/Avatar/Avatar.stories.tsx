import React from 'react';
import Avatar from './Avatar';
import { Bell } from '../icons';

export default {
  title: 'Avatar/Avatar',
  component: Avatar,
};

export const Circle = () => (
  <>
    <Avatar src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/252775/profile-img.jpg" />
    <Avatar style={{ background: '#FE4179', color: 'white' }}>MH</Avatar>
    <Avatar style={{ background: '#1F9DF8', color: 'white' }}>MD</Avatar>
    <Avatar style={{ background: 'var(--greenTeal)', color: 'white' }}>
      <Bell />
    </Avatar>
  </>
);

export const Square = () => (
  <>
    <Avatar
      variant="square"
      src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/252775/profile-img.jpg"
    />
    <Avatar svariant="square" style={{ background: '#FE4179', color: 'white' }}>
      MH
    </Avatar>
    <Avatar variant="square" style={{ background: '#1F9DF8', color: 'white' }}>
      MD
    </Avatar>
    <Avatar
      variant="square"
      style={{ background: 'var(--greenTeal)', color: 'white' }}
    >
      <Bell />
    </Avatar>
  </>
);
