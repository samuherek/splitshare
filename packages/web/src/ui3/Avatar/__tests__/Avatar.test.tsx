import React from 'react';
import { render } from '../../../test-utils/test-utils';
import Avatar, { classes } from '../Avatar';

test('should match snapshots', () => {
  const imgAvatar = render(
    <Avatar src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/252775/profile-img.jpg" />
  );
  expect(imgAvatar.container.firstChild).toMatchSnapshot();

  const textAvatar = render(<Avatar>A</Avatar>);
  expect(textAvatar.container.firstChild).toMatchSnapshot();
});

test('render <Avatar /> correctly', () => {
  const { container } = render(<Avatar />);
  const avatar = container.firstChild;

  expect(avatar).toHaveClass(classes.root);
  // @ts-ignore
  expect(avatar).toHaveProperty('nodeName', 'DIV');
  // @ts-ignore
  expect(avatar).toHaveStyleRule('overflow', 'hidden');
  // @ts-ignore
  expect(avatar).toHaveStyleRule('position', 'relative');
  // @ts-ignore
  expect(avatar).toHaveStyleRule('width', '40px');
  // @ts-ignore
  expect(avatar).toHaveStyleRule('height', '40px');
});

test('prop: variant circle should have correct class and style', () => {
  const { container } = render(<Avatar />);
  const avatar = container.firstChild;

  expect(avatar).toHaveClass(classes.circle);
  // @ts-ignore
  expect(avatar).toHaveStyleRule('border-radius', '50%');
});

test('prop: variant square should have correct class and style', () => {
  const { container } = render(<Avatar variant="square" />);
  const avatar = container.firstChild;

  expect(avatar).toHaveClass(classes.square);
  // @ts-ignore
  expect(avatar).toHaveStyleRule('border-radius', undefined);
});

test('prop: size should apply correct size in px', () => {
  const { container } = render(<Avatar size={20} />);
  const avatar = container.firstChild;

  // @ts-ignore
  expect(avatar).toHaveStyleRule('width', '20px');
  // @ts-ignore
  expect(avatar).toHaveStyleRule('height', '20px');
});

test('prop: component should render provided component as wrapper', () => {
  const { container } = render(<Avatar component="span" />);
  const avatar = container.firstChild;

  expect(avatar).toHaveProperty('nodeName', 'SPAN');
});

test('prop: src should render image if provided', () => {
  const { container } = render(
    <Avatar src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/252775/profile-img.jpg" />
  );
  const img = container.querySelector('img');

  expect(img).toHaveClass(classes.img);
  expect(img).toHaveAttribute('src');
});

test('should render child if child is provided', () => {
  const { getByText } = render(<Avatar>A</Avatar>);
  expect(getByText(/a/i)).toBeDefined();
});

test('should render image if src and child is provided', () => {
  const { container, queryByText } = render(
    <Avatar src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/252775/profile-img.jpg">
      A
    </Avatar>
  );
  const img = container.querySelector('img');

  expect(img).toBeDefined();
  expect(queryByText(/a/i)).toBeNull();
});
