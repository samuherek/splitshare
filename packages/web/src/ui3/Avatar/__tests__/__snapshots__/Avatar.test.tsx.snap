// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
}

.c0 .AvatarImg {
  width: 100%;
  height: 100%;
  object-fit: cover;
  text-align: center;
}

<div
  class="sc-AxjAm c0 Avatar circle"
  size="40"
>
  <img
    class="AvatarImg"
    src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/252775/profile-img.jpg"
  />
</div>
`;

exports[`should match snapshots 2`] = `
.c0 {
  width: 40px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
}

.c0 .AvatarImg {
  width: 100%;
  height: 100%;
  object-fit: cover;
  text-align: center;
}

<div
  class="sc-AxjAm c0 Avatar circle"
  size="40"
>
  A
</div>
`;
